ibm/resource_ibm_atracker_target.go

```diff
@@ -0,0 +1,317 @@
+}
+
+func resourceIBMAtrackerTargetMapToCosEndpoint(cosEndpointMap map[string]interface{}) atrackerv1.CosEndpoint {
+	cosEndpoint := atrackerv1.CosEndpoint{}
+
+	cosEndpoint.Endpoint = core.StringPtr(cosEndpointMap["endpoint"].(string))
+	cosEndpoint.TargetCRN = core.StringPtr(cosEndpointMap["target_crn"].(string))
+	cosEndpoint.Bucket = core.StringPtr(cosEndpointMap["bucket"].(string))
+	cosEndpoint.APIKey = core.StringPtr(cosEndpointMap["api_key"].(string))
+
+	return cosEndpoint
+}
+
+func resourceIBMAtrackerTargetRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	atrackerClient, err := meta.(ClientSession).AtrackerV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getTargetOptions := &atrackerv1.GetTargetOptions{}
+
+	getTargetOptions.SetID(d.Id())
+
+	target, response, err := atrackerClient.GetTargetWithContext(context, getTargetOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetTargetWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("GetTargetWithContext failed %s\n%s", err, response))
+	}
+
+	if err = d.Set("name", target.Name); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting name: %s", err))
+	}
+	if err = d.Set("target_type", target.TargetType); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting target_type: %s", err))
+	}
+	cosEndpointMap := resourceIBMAtrackerTargetCosEndpointToMap(*target.CosEndpoint)
+	if err = d.Set("cos_endpoint", []map[string]interface{}{cosEndpointMap}); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting cos_endpoint: %s", err))
+	}
+	if err = d.Set("crn", target.CRN); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting crn: %s", err))
+	}
+	if err = d.Set("encrypt_key", target.EncryptKey); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting encrypt_key: %s", err))
+	}
+	if target.CosWriteStatus != nil {
+		cosWriteStatusMap := resourceIBMAtrackerTargetCosWriteStatusToMap(*target.CosWriteStatus)
+		if err = d.Set("cos_write_status", []map[string]interface{}{cosWriteStatusMap}); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting cos_write_status: %s", err))
+		}
+	}
+	if err = d.Set("created", dateTimeToString(target.Created)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting created: %s", err))
+	}
+	if err = d.Set("updated", dateTimeToString(target.Updated)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting updated: %s", err))
+	}
+
+	return nil
+}
+
+func resourceIBMAtrackerTargetCosEndpointToMap(cosEndpoint atrackerv1.CosEndpoint) map[string]interface{} {
+	cosEndpointMap := map[string]interface{}{}
+
+	cosEndpointMap["endpoint"] = cosEndpoint.Endpoint
+	cosEndpointMap["target_crn"] = cosEndpoint.TargetCRN
+	cosEndpointMap["bucket"] = cosEndpoint.Bucket
+	cosEndpointMap["api_key"] = cosEndpoint.APIKey
```

Comment:
Can you confirm the API is returning back the value for APIkey 
cosEndpoint.APIKey
Mostly the value is not returned by API...that why the testcase is failing..

Reply:
The APIKey  always returns  with the value "REDACTED"

None of the tests succeeded so there is no target record in the DB.

Example of return data from a POST on a target

```
{
  "id": "8c7b7101-752c-448d-a265-df00ae390437",
  "crn": "crn:v1:staging:public:atracker:us-south:a/64e712e886e178a6f5d7e7cd7dd71f1c::target:8c7b7101-752c-448d-a265-df00ae390437",
  "target_type": "cloud_object_storage",
  "name": "target_name11111",
  "encrypt_key": "REDACTED",
  "cos_endpoint": {
    "endpoint": "s3.au-syd.cloud-object-storage.appdomain.cloud",
    "target_crn": "crn:v1:bluemix:public:cloud-object-storage:global:a/3b76b4a971bfe8e5a1f5baba5d81a845:cd5962f0-943d-46a9-ba7d-7cae757fecb2::",
    "bucket": "atracker-cos-au-syd",
    "api_key": "REDACTED"
  },
  "cos_write_status": {
    "status": "success"
  },
  "created": "2021-07-21T19:33:56.327Z",
  "updated": "2021-07-21T19:33:56.327Z"
}
```